src/Microsoft.DotNet.GitHub.Authentication/GitHubTokenProvider.cs

```diff
@@ -75,22 +105,19 @@ public async Task<string> GetTokenForRepository(string repositoryUrl)
 
     private bool TryGetCachedToken(long installationId, out AccessToken cachedToken)
     {
-        cachedToken = null;
-
-        if (!_tokenCache.ContainsKey(installationId))
+        if (!_tokenCache.TryGetValue(installationId, out cachedToken))
         {
             return false;
         }
 
-        AccessToken token = _tokenCache[installationId];
-
-        // and update the cache
-        if (token.ExpiresAt.Subtract(DateTimeOffset.UtcNow).TotalMinutes < 15)
+        // If the cached token will expire in less than 30 minutes we won't use it,
```

Comment:
```suggestion
        // If the cached token will expire in less than 15 minutes we won't use it,
```
this matches what's in the code, or we can increase the number bellow to 30